src/Riok.Mapperly/Descriptors/MappingBuilders/EnumToStringMappingBuilder.cs

```diff
@@ -64,4 +111,31 @@ private static IReadOnlyDictionary<IFieldSymbol, ExpressionSyntax> BuildExplicit
 
         return explicitMappings;
     }
+
+    private static EnumFallbackValueMapping BuildFallbackMapping(MappingBuilderContext ctx)
+    {
+        var fallbackValue = ctx.Configuration.Enum.FallbackValue;
+        if (fallbackValue is null)
+        {
+            return new EnumFallbackValueMapping(ctx.Source, ctx.Target, new EnumFallbackToStringMapping(ctx.Source, ctx.Target));
+        }
+
+        if (fallbackValue is not { Expression: LiteralExpressionSyntax literalExpressionSyntax })
+        {
+            ctx.ReportDiagnostic(DiagnosticDescriptors.InvalidFallbackValue, fallbackValue.Value.Expression.ToFullString());
+            return new EnumFallbackValueMapping(ctx.Source, ctx.Target, new EnumFallbackToStringMapping(ctx.Source, ctx.Target));
+        }
+
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target, fallbackValue.Value.ConstantValue.Type))
+            return new EnumFallbackValueMapping(ctx.Source, ctx.Target, fallbackExpression: literalExpressionSyntax);
+
+        return new EnumFallbackValueMapping(ctx.Source, ctx.Target, new EnumFallbackToStringMapping(ctx.Source, ctx.Target));
```

Comment:
The target type is always string, isn't it? Couldn't this be simplified to `if (fallbackValue.Value.ConstantValue.Value is not string fallbackStringValue) { diagnostic }` and then construct the literal with `StringLiteral(fallbackStringValue)`?

Reply:
Yes, simplified as suggested.